src/gc.rs

```diff
@@ -201,8 +196,11 @@ impl Collector {
             for addr in (ptr..ptr + size).step_by(WORD_SIZE) {
                 let word = unsafe { *(addr as *const Word) };
 
-                if let Some(ptrinfo) = AllocMetadata::find(word) {
-                    self.worklist.push(ptrinfo)
+                if let Some(block) = ALLOCATOR
+                    .iter()
+                    .find(|x| x.ptr == word || word > x.ptr && word < x.ptr + x.size)
```

Comment:
Should this be `word >= x.ptr`?

Reply:
It can be, I don't think it matters really because if `word == x.ptr` the LHS of the or expression should short circuit.

Reply:
Ah, yeah.